core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/TargetedSequencing.java

```diff
@@ -168,4 +168,34 @@ public void setKitDescriptors(Collection<KitDescriptor> kitDescriptors) {
     this.kitDescriptors.addAll(kitDescriptors);
   }
 
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = 1;
+    result = prime * result + ((alias == null) ? 0 : alias.hashCode());
+    result = prime * result + (archived ? 1231 : 1237);
+    result = prime * result + ((description == null) ? 0 : description.hashCode());
+    result = prime * result + ((targetedSequencingId == null) ? 0 : targetedSequencingId.hashCode());
+    return result;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)